src/main/scala-2.11/org/apache/james/gatling/control/UserFeeder.scala

```diff
@@ -9,22 +9,52 @@ import scala.concurrent.Future
 
 object UserFeeder {
 
+  val USERNAME: String = "username"
+  val PASSWORD: String = "password"
+
   def createUserFeeder(userCount: Int): Array[Map[String, String]] = {
-    val domain = Domain(UUID.randomUUID().toString)
+    toFeeder(createRegisteredUsers(userCount))
+  }
 
+  def createUserFeederWithInboxAndOutbox(userCount: Int): Array[Map[String, String]] = {
+    toFeeder(
+      decorateWithInboxAndOutbox(
+        createRegisteredUsers(userCount)))
+  }
+
+  def decorateWithInboxAndOutbox(users: List[User]): List[User] = {
+    Future.sequence(
+      users.map(user => Future.sequence(
+        List(JamesWebAdministration.createInbox(user.username),
+          JamesWebAdministration.createOutbox(user.username))))
+    ).get
+    users
+  }
+
+  def createRegisteredUsers(userCount: Int): List[User] = {
+    val domain = Domain(UUID.randomUUID().toString)
     JamesWebAdministration.addDomain(domain).get
 
-    val users = (0 until userCount)
+    registerUsers(generateUsers(userCount, domain))
+  }
+
+  def generateUsers(userCount: Int, domain: Domain): List[User] = {
+    (0 until userCount)
       .map(i => User(
         Username(s"""${UUID.randomUUID().toString}@${domain.value}"""),
         Password(UUID.randomUUID().toString)))
       .toList
```

Comment:
Because I prefer a List rather than a TraverseableLike as an argument of other methods, and a traverseable like is not a List :-/